lib/legato/management/account.rb

```diff
@@ -21,16 +21,34 @@ def initialize(attributes, user)
       end
 
       def web_properties
-        WebProperty.for_account(self)
+        @web_properties ||= WebProperty.for_account(self)
+      end
+
+      def web_properties=(value)
+        @web_properties = value
       end
 
       def profiles
-        Profile.for_account(self)
+        if @web_properties
+          @web_properties.inject([]) { |profiles, prop| profiles.concat(prop.profiles) }
+        else
+          Profile.for_account(self)
+        end
       end
 
       def self.from_child(child)
         all(child.user).detect {|a| a.id == child.account_id}
       end
+
+      def self.build_from_summary(attributes, user)
+        properties = attributes['webProperties'] || attributes[:webProperties]
+        account = Account.new(attributes, user)
```

Comment:
You could use `Account.new().tap` here, rather than local variable and then return at the end.